File: model/src/main/java/org/deep_thinker/model/DeepThinkerClient.java
package org.deep_thinker.model;

import java.util.concurrent.CompletableFuture;

public interface DeepThinkerClient {
    CompletableFuture<String> toUpperCase(String s);
    CompletableFuture<Integer> createDQNAgent(DQNConfigFlat config);
    CompletableFuture<Integer> getFirstAction(String agentId, String envId, float[] state);
    CompletableFuture<Integer> getAction(String agentId, String envId, float[] state, float reward);
    CompletableFuture<Integer> episodeComplete(String agentId, String envId, float[] state, float reward, float episodeReward);
}

File: model/src/main/kotlin/org/deep_thinker/model/DeepThinkerClient.kt
package org.deep_thinker.model

import java.util.concurrent.CompletableFuture

interface DeepThinkerClient {
    fun toUpperCase(s: String): CompletableFuture<String>
    fun createDQNAgent(agentId: String, config: DQNConfig): CompletableFuture<String>

    fun getFirstAction(agentId: String, envId: String, state: FloatArray): CompletableFuture<Int>

    fun getAction(agentId: String, envId: String, state: FloatArray, reward: Float): CompletableFuture<Int>

    fun episodeComplete(
        agentId: String, envId: String, state: FloatArray, reward: Float, episodeReward: Float
    ): CompletableFuture<Void>
}